# Examining Racial Discrimination in the US Job Market

### Background
Racial discrimination continues to be pervasive in cultures throughout the world. Researchers examined the level of racial discrimination in the United States labor market by randomly assigning identical résumés to black-sounding or white-sounding names and observing the impact on requests for interviews from employers.

### Data
In the dataset provided, each row represents a resume. The 'race' column has two values, 'b' and 'w', indicating black-sounding and white-sounding. The column 'call' has two values, 1 and 0, indicating whether the resume received a call from employers or not.

Note that the 'b' and 'w' values in race are assigned randomly to the resumes when presented to the employer.

<div class="span5 alert alert-info">
### Exercises
You will perform a statistical analysis to establish whether race has a significant impact on the rate of callbacks for resumes.

Answer the following questions **in this notebook below and submit to your Github account**. 

   1. What test is appropriate for this problem? Does CLT apply?
   2. What are the null and alternate hypotheses?
   3. Compute margin of error, confidence interval, and p-value.
   4. Write a story describing the statistical significance in the context or the original problem.
   5. Does your analysis mean that race/name is the most important factor in callback success? Why or why not? If not, how would you amend your analysis?

You can include written notes in notebook cells using Markdown: 
   - In the control panel at the top, choose Cell > Cell Type > Markdown
   - Markdown syntax: http://nestacms.com/docs/creating-content/markdown-cheat-sheet


#### Resources
+ Experiment information and data source: http://www.povertyactionlab.org/evaluation/discrimination-job-market-united-states
+ Scipy statistical methods: http://docs.scipy.org/doc/scipy/reference/stats.html 
+ Markdown syntax: http://nestacms.com/docs/creating-content/markdown-cheat-sheet
</div>
****

In [1]:
import pandas as pd
import numpy as np
from scipy import stats

In [2]:
data = pd.io.stata.read_stata('data/us_job_market_discrimination.dta')

In [3]:
# number of callbacks for black-sounding names
sum(data[data.race=='b'].call)

157.0

In [4]:
data.head()

id ad  education  ofjobs  yearsexp  honors  volunteer  military  empholes  \
0  b  1          4       2         6       0          0         0         1   
1  b  1          3       3         6       0          1         1         0   
2  b  1          4       1         6       0          0         0         0   
3  b  1          3       4         6       0          1         0         1   
4  b  1          3       3        22       0          0         0         0   

   occupspecific    ...      compreq  orgreq  manuf  transcom  bankreal trade  \
0             17    ...          1.0     0.0    1.0       0.0       0.0   0.0   
1            316    ...          1.0     0.0    1.0       0.0       0.0   0.0   
2             19    ...          1.0     0.0    1.0       0.0       0.0   0.0   
3            313    ...          1.0     0.0    1.0       0.0       0.0   0.0   
4            313    ...          1.0     1.0    0.0       0.0       0.0   0.0   

  busservice othservice  missind  ownership  
0        0.0        0.0      0.0             
1        0.0        0.0      0.0             
2        0.0        0.0      0.0             
3        0.0        0.0      0.0             
4        0.0        1.0      0.0  Nonprofit  

[5 rows x 65 columns]

# What test is appropriate for this problem? Does the CLT apply?

Would like to know the difference in average callback rate between white-sounding names vs. black-sounding names. We are looking for the difference in two means where the population standard deviation is unknown. The sample size is greater than > 30. An appropriate test would be a two tailed T-test comparing the difference of two means. The CLT does apply. 

In [15]:
# H0: mean callback of white-sounding names equals mean callback of black-sounding names
# H1: mean callback of white-sounding names does not equal mean callback of black-sounding names
# significance level: 5%

white_names = data[data.race=='w']
print(str(len(white_names)))

2435


In [17]:
black_names = data[data.race=='b']
print(str(len(black_names)))

2435


In [12]:
white_mean = white_names.call.mean()
print('Mean callback for white-sounding names: ' + str(white_mean))

Mean callback for white-sounding names: 0.09650924056768417


In [14]:
black_mean = black_names.call.mean()
print('Mean callback for black-sounding names: ' + str(black_mean))

Mean callback for black-sounding names: 0.0644763857126236


In [19]:
mean_diff = white_mean - black_mean
print('Difference of mean: ' + str(mean_diff))

Difference of mean: 0.03203285485506058


In [20]:
white_std = white_names.call.std()
print('Callback standard deviation for white-sounding names: ' + str(white_std))

Callback standard deviation for white-sounding names: 0.2953455150127411


In [21]:
black_std = black_names.call.std()
print('Callback standard deviation for white-sounding names: ' + str(black_std))

Callback standard deviation for white-sounding names: 0.24564945697784424


In [22]:
# Margin of error = t critical value * standard error
# standard error = sqrt(white_names callback variance/white_names sample size + black_names callback variance/black_names sample size
# Degrees of freedom: 4871
# t-critical value = 1.96

SE = np.sqrt(white_names.call.var()/len(white_names) + black_names.call.var()/len(black_names))
margin_of_error = 1.96 * SE
print('Standard Error: ' + str(SE))
print('Margin of Error: ' + str(margin_of_error))

Standard Error: 0.00778490682688
Margin of Error: 0.0152584173807


In [24]:
# compute t-statistic

t_stat = mean_diff / SE
print('T-statistic: ' + str(t_stat))

T-statistic: 4.11473837355


In [25]:
# confirm result

stats.ttest_ind(white_names.call, black_names.call)

Ttest_indResult(statistic=4.1147052908617514, pvalue=3.9408021031288859e-05)

Assuming that the mean callback for white-sounding names is equal to the mean callback of black-sounding names, we are 95% Confidence that the difference of means will be +/- 0.015. We received a difference of means 0.032 with the probability of receiving this result well below our significance level of 5%; therefore we reject the null hypothesis that mean callback rates for both are the same.

# Conclusion

This analysis tells us that race/name is a significant factor in callback success. Considering the fact that all other variables were eliminated due to the fact that the two race/name groups had identical resumes; it is clear that the combination of perceived race and name is one of the most important factors in callback success. However our analysis compiles all experience levels into one to receive the net difference. It would be interesting to see whether or not we still receive a statistically significant result when we make the same race/name comparison for all high-experience applicants vs. low-experience applicants. As well as seeing if home location/affluence of neighborhood achieves the same statistcally signifcant result between white-sounding names and black-sounding names.